tailcfg/tailcfg.go

```diff
@@ -36,7 +36,8 @@ import (
 //    10: 2021-01-17: client understands MapResponse.PeerSeenChange
 //    11: 2021-03-03: client understands IPv6, multiple default routes, and goroutine dumping
 //    12: 2021-03-04: client understands PingRequest
-const CurrentMapRequestVersion = 12
+//    13: XXXXX: client understands FilterRule.IPProto
```

Comment:
How do we downgrade a TCP-only ACL for old clients? We'll end up allowing more than the ACL specifies, which isn't a great failure mode.

Reply:
we could wait 6 months until we let users do it without being in a domain-wide feature flag experiment. and those with the feature flag bit will have to acknowledge that they need newer clients?

Reply:
and when we do roll it out, we just don't give those ACL rules to old clients: so they fail closed.

Reply:
Fail closed for ACLs that specify a protocol SGTM. The ACL compiler can be aware of the client's capabilities, and correctly expand both "protocol not specified" (upgrade old rules to rules with explicit protocols listed) and "client doesn't understand protocols" (fail closed).